### PR TITLE
Mock sUSDe Vault Implemented

### PR DESCRIPTION
**Pull request by 0xCSMNT**
Did:
- Implemented a OZ style 4626, same as real sUSDe Vault
- Created a mock USDe token to use as collateral asset for the vault
- Implemented some changes to get the original tests to burn and mint USDB working again

Next Steps:
- More testing same contract
- Implement any changes to harvest() if necessary
- Create updateAccounting() to cache data for read functions on Bestia contracts (nice to have, might add this in when other stuff is in place)

### CODE REVIEW

File: src/USDeVault.sol [hunk 1]
```diff
@@ -18,48 +20,48 @@ contract USDeVault {
...
+    ) {
         assert(_router != address(0));
         assert(_susde != address(0));
         assert(_usdb != address(0));
         assert(_susdb != address(0));
         router = IRouter(_router);
-        susde = IsUSDe(_susde);
-        usde = IERC20(susde.asset());
+        susde = Isusde(_susde);
+        usde = IERC20(_usde);
```
Comment (OxMarco) [inline src/USDeVault.sol:42]: why? just use the ERC4626 native function `asset()` to get the USDe address

File: test/mock/Mock4626Vault.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// Mock ERC4626 Vault for testing
+// Gradually implement functionality of StakedUSDeV2.sol & (inherited) StakedUSDe.sol as needed to match function calls you need
+
+contract Mock4626Vault is ERC4626 {
+    using Math for uint256;
+
+    // STATE VARIABLES
+    uint256 public interestRate;
+    uint256 public constant FACTOR = 1e18;
```
Comment (OxMarco) [inline test/mock/Mock4626Vault.sol:18]: what's this?

File: test/unit/Mock4626Vault.t.sol [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        usde.mint(address(susde), amount);
+
+        console2.log("total USDe in susde", usde.balanceOf(address(susde)));
+        console2.log("totalAssets of susde", susde.totalAssets());
+
+        // asserts that assets in susde accounts for deposits as well as additional USDe minted to susde
+        assertEq(usde.balanceOf(address(susde)), amount * 2);
+    }
+
+    function testConvertToAssets() external {
```
Comment (OxMarco) [inline test/unit/Mock4626Vault.t.sol:44]: these tests are just testing the ERC4626 standard math, I don't think we should keep them

File: src/USDeVault.sol (hunk 1 above)
Comment (0xCSMNT) [reply src/USDeVault.sol:42]: that code did not work. It broke something when you changed to erc4626. this gets it running

File: test/mock/Mock4626Vault.sol (hunk 2 above)
Comment (0xCSMNT) [reply test/mock/Mock4626Vault.sol:18]: I'll delete. not needed

File: test/unit/Mock4626Vault.t.sol (hunk 3 above)
Comment (0xCSMNT) [reply test/unit/Mock4626Vault.t.sol:44]: i'll deleted not needed

**Pull request merged**